File: src/main/kotlin/com/tollwood/company/Company.kt
package com.tollwood.company

import com.tollwood.jpa.Address
import org.hibernate.annotations.CacheConcurrencyStrategy
import javax.persistence.*

@Entity(name = "COMPANY")
@Cacheable
@org.hibernate.annotations.Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
data class Company(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        val id: Long,

        var billNo: Long = 0,
        val name: String,
        val shortName: String,
        var logo: String,
        var thankYouImage: String?,
        @Embedded
        val address: Address,
        val phone: String,
        val email: String,
        val realEstateSupport: Boolean,
        val employeeSupport: Boolean,
        val billingSupport: Boolean,
        val customerSupport: Boolean,
        val publicOrder: Boolean,
        val homeHeader: String,
        val homeFooter: String
)

File: src/main/webapp/src/employees/Company.ts
import Link from "../common/Links";
import {Address} from "../common/Address";
import BankDetails from "./BankDetails";

export default class Company {
    name: string = "";
    shortName: string = "";
    address: Address = new Address();
    billNo: number = 0;
    logo: string = "";
    thankYouImage: string = "";
    phone: string = "";
    email: string = "";
    bankDetails: BankDetails = new BankDetails();
    realEstateSupport: boolean = false;
    employeeSupport: boolean = false;
    statisticSupport: boolean = false;
    billingSupport: boolean = false;
    customerSupport: boolean = false;
    publicOrder : boolean = false;
    homeHeader: string = "";
    homeFooter: string = "";
    _links: {self?: Link} = {};
}